data-prepper-plugins/kinesis-source/src/main/java/org/opensearch/dataprepper/plugins/kinesis/source/processor/KinesisRecordProcessor.java

```diff
@@ -112,7 +114,18 @@ public KinesisRecordProcessor(final BufferAccumulator<Record<Event>> bufferAccum
     }
 
     private KinesisStreamConfig getStreamConfig(final KinesisSourceConfig kinesisSourceConfig) {
-        final Optional<KinesisStreamConfig> kinesisStreamConfig = kinesisSourceConfig.getStreams().stream().filter(streamConfig -> streamConfig.getName().equals(streamIdentifier.streamName())).findAny();
+        final Optional<KinesisStreamConfig> kinesisStreamConfig = kinesisSourceConfig.getStreams().stream().filter(streamConfig -> {
+            if (streamIdentifier.streamArnOptional().isPresent()) {
+                Arn streamIdentifierArn = streamIdentifier.streamArnOptional().get();
+                Arn streamConfigArn = Arn.fromString(streamConfig.getArn());
+                return streamIdentifierArn.equals(streamConfigArn);
+            }
+            if (Objects.nonNull(streamConfig.getArn())) {
+                String streamName = Arn.fromString(streamConfig.getArn()).resource().resource();
+                return streamName.equals(streamIdentifier.streamName());
+            }
+            return streamConfig.getName().equals(streamIdentifier.streamName());
+        }).findAny();
```

Comment:
This `findAny` can return one if there are multiple matching. Should we have some check to ensure there is only one?